### PR TITLE
PP-8460 store 3DS version info for Stripe payments

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- For Stripe payments that go through 3DS journey, record 3DS version available from Stripe.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java [hunk 1]
```diff
@@ -263,20 +264,25 @@ private void processSourceNotification(StripeNotification notification) {
...
+                executePost3DSAuthorisation(charge, notification.getType(), Optional.empty());
             }
 
         } catch (StripeParseException e) {
             logger.error("{} notification parsing for source object failed: {}", PAYMENT_GATEWAY_NAME, e);
         }
     }
 
-    private void executePost3DSAuthorisation(ChargeEntity charge, String notificationEventType) {
+    private void executePost3DSAuthorisation(ChargeEntity charge, String notificationEventType, Optional<StripeCharge> optionalStripeCharge) {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java]: Our coding standards say that we shouldn't use Optionals as method parameters https://pay-team-manual.cloudapps.digital/manual/coding-standards/java-coding-standards.html#optionals

I think we should always have the charge on the payment intent for this kind of notification. So we could possibly get it in the calling method, else throw an exception. Or pass in the whole `PaymentIntent` to this method?

**Review by stephencdaly — COMMENTED** [review]
Looks good. Left a comment about optionals. Also, are there any integration tests we could add/modify to ensure the 3ds version is persisted on the charge?

File: src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java (hunk 1 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java]: 👍

File: src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java [hunk 2]
```diff
@@ -263,21 +265,27 @@ private void processSourceNotification(StripeNotification notification) {
...
 
             Auth3dsResult auth3DsResult = new Auth3dsResult();
             auth3DsResult.setAuth3dsResult(getMappedAuth3dsResult(type));
 
+            Optional<StripeCharge> stripeCharge = paymentIntent.getCharge();
+            if (stripeCharge.isPresent() && stripeCharge.get().getPaymentMethodDetails() != null &&
+                    stripeCharge.get().getPaymentMethodDetails().getThreeDSecure() != null) {
+                    auth3DsResult.setThreeDsVersion(stripeCharge.get().getPaymentMethodDetails().getThreeDSecure().getVersion());
+            }
+
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/StripeNotificationService.java]: Could use `ifPresent` so save having to call `get()` multiple times.

```suggestion
            paymentIntent.getCharge().ifPresent(stripeCharge -> {
                if (stripeCharge.getPaymentMethodDetails() != null &&
                        stripeCharge.getPaymentMethodDetails().getThreeDSecure() != null) {
                    auth3DsResult.setThreeDsVersion(stripeCharge.getPaymentMethodDetails().getThreeDSecure().getVersion());
                }
            });
```

**Review by stephencdaly — APPROVED** [review]
LGTM!

**Pull request merged**